AudioManagerUtilities/src/CAmSocketHandler.cpp

```diff
@@ -423,19 +457,35 @@ am::am_Error_e CAmSocketHandler::addFDPoll(const int fd, const short event, IAmS
   */
 am_Error_e CAmSocketHandler::removeFDPoll(const sh_pollHandle_t handle)
 {
+    CHECK_CALLER_THREAD_ID()
+    
     VectorListPoll_t::iterator iterator = mListPoll.begin();
-
+    
     for (; iterator != mListPoll.end(); ++iterator)
     {
         if (iterator->handle == handle)
         {
             iterator = mListPoll.erase(iterator);
             mSetPollKeys.pollHandles.erase(handle);
             mRecreatePollfds = true;
-            return (E_OK);
+            break;
         }
     }
-    return (E_UNKNOWN);
+    
+    if (iterator == mListPoll.end())
+        return (E_UNKNOWN);
```

Comment:
Looks OK I think?  iterator points to end() only if the handle was never found in the list, doesn't it?  By convention end() is one beyond the last element, right? 
.

Reply:
I fear the iterator will point to end() _also_ if the handle is found as last element of the list.
Following code should mimic the situation, trying to specify "last" as argument would lead to the unwanted case:
```
#include <iostream>
#include <vector>
#include <string>

int main(int argc, char **argv)
{
        if (argc != 2)
        {
                std::cout << "Launch with one argument which is erasure pattern" << std::endl;
                return -1;
        }   
        std::vector<std::string> v = {"foo", "bar", "last"};
        auto it = v.begin();
        for (; it != v.end(); ++it)
        {
                if (it->compare(argv[1]) == 0)
                {   
                        it = v.erase(it);
                        break;
                }   
        }   
        if (it == v.end())
        {   
                std::cout << "Watch out, we are returning UNKNOWN!" << std::endl;
                return 0;
        }   
        std::cout << "Everything was fine" << std::endl;
}
```

Reply:
Sorry I totally missed that the loop was doing erase().  Good catch.

Reply:
Solution can be to change just one line:
it = v.erase(it); -> v.erase(it);